test/harry/main/org/apache/cassandra/harry/model/ASTSingleTableModel.java

```diff
@@ -0,0 +1,1289 @@
+    {
+        if (columns.isEmpty()) return Collections.singletonList(Clustering.EMPTY);
+        List<ByteBuffer[]> current = new ArrayList<>();
+        current.add(new ByteBuffer[columns.size()]);
+        for (Symbol symbol : columns)
+        {
+            int position = columns.indexOf(symbol);
+            List<? extends Expression> expressions = values.get(symbol);
+            ByteBuffer firstBB = eval(expressions.get(0));
+            current.forEach(bbs -> bbs[position] = firstBB);
+            if (expressions.size() > 1)
+            {
+                // this has a multiplying effect... if there is 1 row and there are 2 expressions, then we have 2 rows
+                // if there are 2 rows and 2 expressions, we have 4 rows... and so on...
+                List<ByteBuffer[]> copy = new ArrayList<>(current);
+                for (int i = 1; i < expressions.size(); i++)
+                {
+                    ByteBuffer bb = eval(expressions.get(i));
+                    for (ByteBuffer[] bbs : copy)
+                    {
+                        bbs = bbs.clone();
+                        bbs[position] = bb;
+                        current.add(bbs);
+                    }
+                }
+            }
+        }
+        return current.stream().map(BufferClustering::new).collect(Collectors.toList());
+    }
+
+    private static ByteBuffer eval(Expression e)
+    {
+        return ExpressionEvaluator.tryEvalEncoded(e).get();
+    }
+
+    private static class Row
+    {
+        private final ImmutableUniqueList<Symbol> columns;
+        private final ByteBuffer[] values;
+
+        private Row(ImmutableUniqueList<Symbol> columns, ByteBuffer[] values)
+        {
+            this.columns = columns;
+            this.values = values;
+        }
+
+        public String asCQL(Symbol symbol)
+        {
+            int offset = columns.indexOf(symbol);
+            assert offset >= 0;
+            ByteBuffer b = values[offset];
+            return (b == null || ByteBufferUtil.EMPTY_BYTE_BUFFER.equals(b)) ? "null" : symbol.type().asCQL3Type().toCQLLiteral(b);
+        }
+
+        public List<String> asCQL()
+        {
+            List<String> human = new ArrayList<>(values.length);
+            for (int i = 0; i < values.length; i++)
+                human.add(asCQL(columns.get(i)));
+            return human;
+        }
+
+        @Override
+        public boolean equals(Object o)
+        {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            Row row = (Row) o;
+            return Arrays.equals(values, row.values);
+        }
+
+        @Override
+        public int hashCode()
+        {
+            return Arrays.hashCode(values);
+        }
+
+        @Override
+        public String toString()
+        {
+            return asCQL().toString();
+        }
+    }
+
+    private class LookupContext
+    {
+        private final Select select;
```

Comment:
Happens to be unused

Reply:
i had this for when i was working in a debugger... but in all cases i can just move higher up in the stack trace, so not needed